libraries/ui-library-angular/src/lib/form/six-form.directive.ts

```diff
@@ -0,0 +1,111 @@
+import { Directive, ElementRef, EventEmitter, HostListener, Injector, Output } from '@angular/core';
+import { AbstractControl, FormArray, FormControl, FormGroup, FormGroupDirective } from '@angular/forms';
+
+/**
+ * This directive intercepts the ngSubmit event of an Angular form and introduces
+ * a supplementary event named sixSubmit. The sixSubmit event is triggered exclusively
+ * when the form is valid. In cases where the form is considered invalid, this directive
+ * takes proactive actions by marking all form controls as touched and dirty. Additionally,
+ * it shifts the focus to the initial invalid form element, facilitating quick error
+ * resolution.
+ *
+ * To utilize this directive, apply it to an Angular form.
+ * ```html
+ * <form [formGroup]="form" sixForm (sixSubmit)="onSubmit($event)">
+ *   <!-- form content -->
+ * </form>
+ * ```
+ *
+ * For users needing greater flexibility in determining when error messages are displayed,
+ * or for those who prefer not to rely solely on the form submission event,
+ * an alternative is to use the SixFormUtilDirective.
+ */
+@Directive({
+  selector: 'form[sixForm]',
+})
+export class SixFormDirective {
+  /**
+   * Emits an event when the form is valid and the form submission has been triggered.
+   */
+  @Output() sixSubmit = new EventEmitter<SubmitEvent>();
+
+  @HostListener('ngSubmit', ['$event'])
+  onNgSubmit(event: SubmitEvent): void {
+    const formGroupDirective = this.injector.get(FormGroupDirective);
```

Comment:
Couldn't this be undefined iff it is optional below?

Reply:
Yes, but now that it is [injected directly](https://github.com/six-group/six-webcomponents/pull/81/files/6d70bba773c2248b911e88fec74c4276a91276da#r1295579590), it should never be undefined.